instrumentation/aws-lambda/aws-lambda-core-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambdacore/v1_0/internal/CarrierEnricher.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awslambdacore.v1_0.internal;
+
+import java.util.Map;
+
+/**
+ * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+ * any time.
+ */
+public interface CarrierEnricher {
```

Comment:
I'd suggest removing the interface since there's only a single implementation (and since it's internal)

Reply:
My intention was to show that the behaviour could be abstracted and new implementations could exist in the future. I will remove it to simplify the POC.,